go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -220,13 +221,37 @@ func (n *NodeController) hybridOverlayNodeUpdate(node *kapi.Node) error {
 			"set_field:%s->eth_dst,"+
 			"output:"+extVXLANName,
 			cookie, cidr.String(), n.gwLRPIP.String(), hotypes.HybridOverlayVNI, n.drIP, nodeIP.String(), drMAC.String()))
+
+	if len(config.HybridOverlay.ClusterSubnets) == 0 {
+		// No static cluster subnet is provided in config. Try to detect the hybrid overlay node subnet dynamically
+		// Add a route via the hybrid overlay port IP through the management port
+		// interface for each hybrid overlay cluster subnet
+		mgmtPortLink, err := netlink.LinkByName(types.K8sMgmtIntfName)
+		if err != nil {
+			return fmt.Errorf("failed to lookup link %s: %v", types.K8sMgmtIntfName, err)
+		}
+
+		route := &netlink.Route{
```

Comment:
why is this route necessary? host to HO subnet access? Wouldn't we also need this for regular HO?

Reply:
Yes, this is for the host to HO subnet traffic. We have a similar route added for regular HO at https://github.com/ovn-org/ovn-kubernetes/blob/fc66d17e7753f601489964d6c24a00315c36b4bc/go-controller/hybrid-overlay/pkg/controller/node_linux.go#L438.

In regular HO, we have it in `EnsureHybridOverlayBridge`. Because the HO subnet is from the configuration, the route can be configured when a HO node is added to the cluster. But when `HybridOverlay.ClusterSubnets` is not present, we need to get the HO subnets from the annotation of each HO node.  So I put the logic here.